test/test_crawl.py

```diff
@@ -42,117 +41,99 @@ def get_public_suffix(url):
+    visited sites (with a few exceptions). If it does not, it is likely
+    the profile is lost at some point during the crawl.
     """
-
-    def get_config(
-        self, data_dir: Path = None
-    ) -> Tuple[ManagerParams, List[BrowserParams]]:
-        manager_params, browser_params = self.get_test_config(data_dir)
-        browser_params[0].profile_archive_dir = os.path.join(
-            manager_params.data_directory, "browser_profile"
-        )
-        browser_params[0].http_instrument = True
-        return manager_params, browser_params
-
-    @pytest.mark.xfail(run=False)
-    @pytest.mark.slow
-    def test_browser_profile_coverage(self, tmpdir: Path, task_manager_creator) -> None:
-        """Test the coverage of the browser's profile
-
-        This verifies that Firefox's places.sqlite database contains
-        all visited sites (with a few exceptions). If it does not,
-        it is likely the profile is lost at some point during the crawl
-        """
-        # Run the test crawl
-        data_dir = tmpdir / "data_dir"
-        manager_params, browser_params = self.get_config(data_dir)
-        manager, crawl_db = task_manager_creator((manager_params, browser_params))
-        for site in TEST_SITES:
-            manager.get(site)
-        ff_db_tar = os.path.join(
-            browser_params[0].profile_archive_dir, "profile.tar.gz"
+    # Run the test crawl
+    manager_params, browser_params = default_params
+    manager_params.num_browsers = 1
+    browser_params[0].profile_archive_dir = (
+        manager_params.data_directory / "browser_profile"
+    )
+    browser_params[0].http_instrument = True
+    manager, crawl_db = task_manager_creator((manager_params, browser_params[:1]))
+    for site in TEST_SITES:
+        manager.get(site)
+    manager.close()
+
+    # Extract crawl profile
+    ff_db_tar = browser_params[0].profile_archive_dir / "profile.tar.gz"
+    with tarfile.open(ff_db_tar) as tar:
+        tar.extractall(browser_params[0].profile_archive_dir)
+
+    # Output databases
+    ff_db = browser_params[0].profile_archive_dir / "places.sqlite"
+
+    # Grab urls from crawl database
+    rows = db_utils.query_db(crawl_db, "SELECT url FROM http_requests")
+    req_ps = set()  # visited domains from http_requests table
+    for (url,) in rows:
+        req_ps.add(get_public_suffix(url))
+
+    hist_ps = set()  # visited domains from crawl_history Table
+    statuses = dict()
+    rows = db_utils.query_db(
+        crawl_db,
+        "SELECT arguments, command_status FROM crawl_history WHERE command='GET'",
+    )
+    for url, command_status in rows:
+        ps = get_public_suffix(url)
+        hist_ps.add(ps)
+        statuses[ps] = command_status
+
+    # Grab urls from Firefox database
+    profile_ps = set()  # visited domains from firefox profile
+    rows = db_utils.query_db(ff_db, "SELECT url FROM moz_places")
+    for (host,) in rows:
+        try:
+            profile_ps.add(get_public_suffix(host))
```

Comment:
Is this actually adding the public suffix (e.g., `com`) or the public suffix + 1 (e.g., `example.com`)? If it's the former, it seems like we should update this to use public suffix + 1. To grab that we can directly use `du.get_ps_plus_1`.

Reply:
Good catch! This was adding the publix suffix indeed so I changed it to use `du.get_ps_plus_1`.
Also, after looking at the test more closely, I noticed that one of the queries to the crawl database was not formed correctly (see 9e8298c).